megamek/src/megamek/client/ui/swing/unitDisplay/UnitDisplay.java

```diff
@@ -68,41 +83,197 @@ public UnitDisplay(@Nullable ClientGUI clientgui,
+        c.gridwidth = GridBagConstraints.REMAINDER;
+
+        ((GridBagLayout) getLayout()).setConstraints(butSwitchView, c);
+        add(butSwitchView);
+
+        butSwitchView.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (!(GUIPreferences.getInstance().getDisplayStartTabbed())) {
+                    setDisplayTabbed();
+                }
+                else {
+                    setDisplayNonTabbed();
+                }
+
+                displayP.revalidate();
+            }
+        });
+
+        if (GUIPreferences.getInstance().getDisplayStartTabbed()) {
+            setDisplayTabbed();
+        }
+        else {
+            setDisplayNonTabbed();
+        }
+    }
+
+    /**
+     * switch display to the tabbed version
+     *
+     */
+    private void setDisplayTabbed() {
+        tabStrip.setVisible(true);
+
+        if (displayP.getComponentCount() > 0){
+            displayP.removeAll();
+        }
+        if (mPanP.getComponentCount() > 0){
+            mPanP.removeAll();
+        }
+        if (pPanP.getComponentCount() > 0){
+            pPanP.removeAll();
+        }
+        if (aPanP.getComponentCount() > 0){
+            aPanP.removeAll();
+        }
+        if (wPanP.getComponentCount() > 0){
+            wPanP.removeAll();
+        }
+        if (sPanP.getComponentCount() > 0){
+            sPanP.removeAll();
+        }
+        if (ePanP.getComponentCount() > 0){
+            ePanP.removeAll();
+        }
+
+        displayP.add(MechPanelTabStrip.SUMMARY, mPan);
+        displayP.add(MechPanelTabStrip.PILOT, pPan);
+        displayP.add(MechPanelTabStrip.ARMOR, aPan);
+        displayP.add(MechPanelTabStrip.WEAPONS, wPan);
+        displayP.add(MechPanelTabStrip.SYSTEMS, sPan);
+        displayP.add(MechPanelTabStrip.EXTRAS, ePan);
+
+        tabStrip.setTab(MechPanelTabStrip.SUMMARY_INDEX);
+
+        GUIPreferences.getInstance().setDisplayStartTabbed(true);
+    }
+
+    /**
+     * switch display to the non tabbed version
+     *
+     */
+    private void setDisplayNonTabbed() {
+        tabStrip.setVisible(false);
+
+        if (displayP.getComponentCount() > 0){
+            displayP.removeAll();
+        }
+
+        mPan.setVisible(true);
+        pPan.setVisible(true);
+        aPan.setVisible(true);
+        wPan.setVisible(true);
+        sPan.setVisible(true);
+        ePan.setVisible(true);
+
+        mPanP.add(mPan, BorderLayout.CENTER);
+        pPanP.add(pPan, BorderLayout.PAGE_START);
+        aPanP.add(aPan, BorderLayout.CENTER);
+        wPanP.add(wPan, BorderLayout.CENTER);
+        sPanP.add(sPan, BorderLayout.CENTER);
+        ePanP.add(ePan, BorderLayout.CENTER);
+        displayP.add(splitABC);
+
+        GUIPreferences.getInstance().setDisplayStartTabbed(false);
+    }
+
+    /*
+    * Save splitter locations to preferences
+    *
+    */
+    public void saveSpliterLoc() {
```

Comment:
Hate to be that guy, but can we correct the method's spelling to "saveSplitterLoc"?